jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorPageErrorProcessor.java

```diff
@@ -0,0 +1,115 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.URIUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An ErrorProcessor that can re-handle a request at an error page location.
+ */
+public abstract class ErrorPageErrorProcessor extends ErrorProcessor
+{
+    private static final Logger LOG = LoggerFactory.getLogger(ErrorPageErrorProcessor.class);
+
+    private final Handler _handler;
+
+    protected ErrorPageErrorProcessor(Handler handler)
+    {
+        _handler = handler;
+    }
+
+    @Override
+    public InvocationType getInvocationType()
+    {
+        return _handler.getInvocationType();
+    }
+
+    @Override
+    protected void generateAcceptableResponse(Request request, Response response, int code, String message, Throwable cause, Callback callback) throws IOException
```

Comment:
Given this new implementation, I would rename this method of the base class, since it does not deal with the `Accept` header, to just `generateResponse()`.